src/main/java/jenkins/plugins/logstash/LogstashBuildWrapper.java

```diff
@@ -67,7 +76,26 @@ public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener l
   public OutputStream decorateLogger(AbstractBuild build, OutputStream logger) {
     LogstashWriter logstash = getLogStashWriter(build, logger);
 
-    return new LogstashOutputStream(logger, logstash);
+    LogstashOutputStream los = new LogstashOutputStream(logger, logstash);
+
+    if (build.getProject() instanceof BuildableItemWithBuildWrappers) {
+      BuildableItemWithBuildWrappers project = (BuildableItemWithBuildWrappers) build.getProject();
+      for (BuildWrapper wrapper: project.getBuildWrappersList()) {
+        if (wrapper instanceof MaskPasswordsBuildWrapper) {
+          List<VarPasswordPair> allPasswordPairs = new ArrayList<VarPasswordPair>();
+
+          MaskPasswordsBuildWrapper maskPasswordsWrapper = (MaskPasswordsBuildWrapper) wrapper;
+          allPasswordPairs.addAll(maskPasswordsWrapper.getVarPasswordPairs());
+
+          MaskPasswordsConfig config = MaskPasswordsConfig.getInstance();
+          allPasswordPairs.addAll(config.getGlobalVarPasswordPairs());
+
+          return (OutputStream) los.maskPasswords(allPasswordPairs);
```

Comment:
MaskPasswordsOutputStream is a descendant of OutputStream, so you don't need the cast here.

`return los.maskPasswords(allPasswordPairs);`